ts/components/messages/MessagesDeadlines.tsx

```diff
@@ -75,6 +75,27 @@ type State = {
   maybeLastLoadedStartOfMonthTime: Option<number>;
   lastMessagesState?: pot.Pot<ReadonlyArray<MessageState>, string>;
   allMessageIdsState: Set<string>;
+  finalDeadlineId: string;
+};
+
+/**
+ * Get the last deadline, after order the sections the older is in position 0
+ */
+const getFinalDeadlineId = (sections: Sections) => {
```

Comment:
This could be rewritten as follow

```typescript
/**
 * Get the last deadline id (the oldest in time is the first in array position)
 */
const getLastDeadlineId = (sections: Sections): Option<string> => {
  return fromNullable(sections)
    .chain(s => fpIndex(0, s))
    .chain(d => fpIndex(0, [...d.data]))
    .fold(none, item => {
      if (!isFakeItem(item)) {
        return some(item.e1.id);
      }
      return none;
    });
};
```
where **fpIndex** is 
`import { index as fpIndex } from "fp-ts/lib/Array";`

So I guess it's more representative to return an **Option<string>**: the id could be or not.